### DISCUSSION TITLE
MiXCR Default Library

### DISCUSSION THREAD

**Discussion by bshim181** [milaboratory/mixcr#1588] [Q&A] [answered]
Hello, 

I had a question regarding the MiXCR's defaut library reference sequences. I noticed there were some differences in terms of list of alleles for specific v genes and what v genes were included in the reference sequences. Could you elaborate little more on how the default library was built?

Also, if i would want to compare the scTCR output from cell ranger with bulk TCR output from MiXCR to do clonotype expansion tracking over time, how would you recommend synchronizing the reference sequences?

**Comment by mizraelson** [comment, marked as answer]
1) The MiXCR library is based on genomic information and does not include alleles directly. To infer alleles from the data, there is a dedicated function,[`mixcr findAlleles` ](https://mixcr.com/mixcr/reference/mixcr-findAlleles/). This method is considered more reliable because, in many other databases, hypermutations and sequencing/PCR errors are often mistakenly identified as alleles. 

2) I recommend comparing different approaches by mapping the actual VDJ region nucleotide sequence, as the sequence should remain consistent regardless of the allele identification.

**Reply by bshim181** [reply]
<img width="1023" alt="Screenshot 2024-03-13 at 3 03 43 PM" src="https://github.com/milaboratory/mixcr/assets/53489568/78924643-06c4-4ddf-9c27-3a8b84488919">

I was looking at the VDJ online sequences, assuming that these were the reference sequences utilized for MiXCR's default libraries. I took these reference sequences and generated a custom reference file compatible with Cellranger run.  Then I compared the outputs generated while using the custom reference vs Cellranger's imported IMGT reference file. 

<img width="962" alt="Screenshot 2024-03-13 at 3 07 18 PM" src="https://github.com/milaboratory/mixcr/assets/53489568/3286ed92-fbb0-4d94-8c48-4a2e7fefd0d4">

There were some level of differences in how v genes were called for each clonotypes between outputs. That is why I was trying to better understand how the default libraries are built. I guess I am still confused when you say "The MiXCR library is based on genomic information and does not include alleles directly." In that case, are list of V genes and V gene allele sequences in VDJ online database not used directly during MiXCR alignments?

**Reply by mizraelson** [reply]
Yes, in the MiXCR build in library all genes have *00 allele. The alleles are only inferred from the data itself with `mixcr findAlleles` (the algorithm uses the database of alleles you see in vdj.online but each time the presence of a certain variant has to be supported by the data). Technically speaking the *00 allele is the one that is present in the genome assembly the library is based on.
